src/main/scala/shine/C/AST/Printer.scala

```diff
@@ -264,10 +264,13 @@ class CPrinter extends Printer {
   private def printFunCall(f: FunCall): Unit = {
     printDeclRef(f.fun)
     print("(")
-    f.args.foreach(a => {
-      printExpr(a)
-      if (!a.eq(f.args.last)) print(", ")
-    })
+    if (f.args.nonEmpty) {
+      f.args.take(f.args.length - 1).foreach(a => {
+        printExpr(a)
+        print(", ")
+      })
+      printExpr(f.args.last)
+    }
```

Comment:
Is this not doing exactly the same as before, just with more code?

Reply:
The previous version did not work if the last argument is also passed as first argument, cause the `,` is only printed when the current argument not equals the last argument. So if the first and the last argument are the same, there is missing one `,`.

Reply:
ok.